src/Kafka/Internal/Setup.hs

```diff
@@ -15,7 +16,7 @@ import Kafka.Internal.CancellationToken
 newtype KafkaProps = KafkaProps [(String, String)] deriving (Show, Eq)
 newtype TopicProps = TopicProps [(String, String)] deriving (Show, Eq)
 newtype Kafka      = Kafka RdKafkaTPtr deriving Show
-data KafkaConf     = KafkaConf RdKafkaConfTPtr CancellationToken
+data KafkaConf     = KafkaConf RdKafkaConfTPtr (IORef (Maybe RdKafkaQueueTPtr)) CancellationToken
```

Comment:
Unfortunately, the new queue should be in `IORef`. This is because:
- the new queue must be used in a callback
- callbacks must be set on `KafkaConf` _before_ `Kafka` handle is created
- `KafkaConf` is needed in order to create `Kafka` handle
- `Kafka` handle is required in order to create a new queue.

Therefore we create `KafkaConf`, set all the callbacks, create `Kafka` handle, create the queue and make it available for the callbacks via IORef.